File: testdata/repos/python-fastapi-clean/services/user_service.py
from repositories.user_repo import UserRepository

class UserService:
    def __init__(self):
        self.repo = UserRepository()

    def get_user(self, user_id: str) -> dict:
        return self.repo.find_by_id(user_id)

    def create_user(self, data: dict) -> dict:
        return self.repo.create(data)

File: testdata/benchmark/trusted-core/ARCH-PATTERN-001/pass/service/user_service.go
package service

// UserRepo defines the repository interface for user operations.
type UserRepo interface {
	FindByID(id string) (map[string]string, error)
	Save(user map[string]string) error
}

// UserService handles business logic, delegating data access to the repository.
type UserService struct {
	repo UserRepo
}

// NewUserService creates a new UserService with the given repository.
func NewUserService(repo UserRepo) *UserService {
	return &UserService{repo: repo}
}

// GetUser retrieves a user by ID.
func (s *UserService) GetUser(id string) (map[string]string, error) {
	return s.repo.FindByID(id)
}
